### DISCUSSION TITLE
Predefined "Block Definitions" in the Block Editor Family

### DISCUSSION THREAD

**Discussion by royberris** [umbraco/Umbraco-CMS#12629] [Features and ideas]
Hi, I want to propose an idea for the block editor family where we can predefine a "block definition". (Not sure if that is the name). 

So now we can create a blocklist editor, and define blocks which are a set of properties like the content element type, the settings element type a name and some other settings. 

If I would have page 1 and page 2. Each having their blocklist editor to restrict editors from placing any block. And I would allow page 1 to have blocks A, B, C and page 2 to have blocks B, C and D. I now need to redefine blocks B and C for said blocklist editors. Now imagine this with 20 page doctypes and 30 blocks.

It would be nice if we could get a new data type (maybe called a "block definition"). Which would allow us to define blocks that span multiple blocklists. This will just be the same as creating it 'inline', but predefined. Now we can reference this block definition from multiple block editors. And of course when we would change the definition it would persist through all block editors. 

With the upcoming Block Grid this could allow for us to easily share blocks between different types of Block Editors. 
Let me know what you think. 

(I got told to tag you @nielslyngsoe )

**Comment by nielslyngsoe** [comment]
I love this idea. Similar other ideas have been presented to me over time and I agree that this matter should be dealt with.

As much as I love it, Its just not something we can react on currently as we are prioritizing 'New Backoffice' (https://github.com/umbraco/rfcs/discussions/34)
I have raised awareness about this missing feature, so it hopefully can become a priority in the future.

Feel free to provide more concrete proposals and even get started on such feature in a PR.

Thanks

**Reply by royberris** [reply]
Will definitely attempt a PR for this. I'm not too familiar with the umbraco source code at this time, but it might be a nice way to start learning it.
